### ISSUE TITLE
how do I bind express-hbs templating engine?

### ISSUE THREAD

**Issue by shukriadams** [keystonejs/keystone#4514]
I'm trying to switch the templating engine to https://github.com/barc/express-hbs.

    hbs = require('express-hbs');

    hbs.express4({
        helpers: new require('./templates/helpers')(),
        layoutsDir: 'templates/layouts',
        partialsDir: 'templates/partials',
        defaultLayout: 'default.hbs',
        extname: '.hbs'
    });

    keystone.init({
        'custom engine': hbs.express4,
    });

this starts the keystone app, but I get an error in express-hbs "Cannot create property 'extname' on string...." for whatever template page it's trying to render, because keystone or express are passing a string to express-hbs when it expects an object. So I'm guessing it's because I haven't bound express-hbs correctly, but there's not a whole lot of documentation showing how to bind other Express-compliant engines.

**Comment by shukriadams** [comment]
@CAYdenberg Thanks. Unfortunately "express-handlebars" is the engine I'm trying to get away from right now, I can't get more advanced template rendering to work in it and the project seems to be abandoned - there are a lot of unanswered render-related tickets on their project. I'm trying to hook other engines up to Keystone and see if I have better luck with those.

It doesn't look like I can pass express-hbs to Keystone the same way I can with express-handlebars, but express-hbs seems to an Express-compatible engine, so I'm wondering how the Keystone devs suggest we connect a random Express engine to Keystone.

**Comment by CAYdenberg** [comment]
Yeah it is kind of weird. I would have assumed that 'custom engine' just plugs into the express engine interface.

**Comment by htor** [comment]
>... so I'm wondering how the Keystone devs suggest we connect a random Express engine to Keystone.

according to docs, in `keystone.js` you do:
```
const express = require('express')
const app = express()
keystone.set('app', app)
...
keystone.start()
```
but then when starting keystone the the routes are not set up on the provided `app`. i never got it to work with my own express instance.
